hyperspy/drawing/widgets.py

```diff
@@ -525,11 +527,14 @@ def update_range(self):
                       self.rect.get_x() + self.rect.get_width())
 
     def move_left(self, event):
-        if self.buttonDown is False or self.ignore(event):
+        if self.buttonDown is False or self.ignore(event) or\
+                event.xdata is None:
             return
         # Do not move the left edge beyond the right one.
         if event.xdata >= self.range[1]:
             return
+        if self.left and event.xdata < self.left:
+            return
```

Comment:
Good point. I've fixed it without adding the extra method.


On Mon, 2016-02-15 at 09:16 -0800, Vidar Tonaas Fauske wrote:
> In hyperspy/drawing/widgets.py:
> >              return
> >          # Do not move the left edge beyond the right one.
> >          if event.xdata >= self.range[1]:
> >              return
> > +        if self.left and event.xdata < self.left:
> > +            return
> In general, I would make a function to bound the xdata:
> def _validate_pos(self, x):
>     if self.left is not None:
>         x = np.max(x, self.left)
>     if self.right is not None:
>         x = np.min(x, self.right)
>     return x
> This then sets x to the appropriate limit if it is out of bounds.
> This is better since we're never guaranteed to have an event for each
> pixel moved. To save some events, it might also benefit from checking
> whether the value actually changes.
> —
> Reply to this email directly or view it on GitHub.